src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1797,6 +1797,13 @@
           (create-extract-node* cols expr))
       (create-extract-node* query-rec cols expr))))
 
+(defn try-parse-timestamp
+  "Try to convert a string to a timestamp, throwing an exception if it fails"
+  [ts]
+  (if-let [result (to-timestamp ts)]
+    result
+    (throw (IllegalArgumentException. (tru "''{0}'' is not a valid timestamp value" ts)))))
```

Comment:
Just noticed we could also write this as:
```
(or (to-timestamp ts)
    (throw ....)
```
But if you have a preference, that's fine.